File: Data Structures & Algorithms/string-encode-and-decode/submission-0.java
class Solution {

    public String encode(List<String> strs) {
        StringBuilder sb = new StringBuilder();
        for(String s : strs){
            for(char c : s.toCharArray()){
                if(c == (char)0){
                    sb.append((char)255);
                }else{
                    sb.append((char)(c - 1));
                }
            }
            sb.append((char) 0);
        }
        return sb.toString();
    }

    public List<String> decode(String str) {
        List<String> novaLista = new ArrayList<>();
        StringBuilder sb =new StringBuilder();
        for(char c : str.toCharArray()){
            if(c == 255){
                sb.append((char) 0);
            }else if(c == 0){
                novaLista.add(sb.toString());
                sb.setLength(0);
            }else{
                sb.append((char)(c + 1));
            }
        }
        return novaLista;
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-0.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        val sb = StringBuilder()
        var i = 0;
        while(true){
            val ch = try{
                strs[0][i]
            }catch(_ : IndexOutOfBoundsException){
                break;
            }
            for(j in 1 until strs.size){
                if(strs[j].length <= i || strs[j][i] != ch){
                    return sb.toString()
                }
            }
            i++
            sb.append(ch)
        }

        return sb.toString()
    }
}
